workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -212,9 +214,107 @@ describe('DynamicAuthorizationService', () => {
   });
 
   describe('deleteGroup', () => {
-    it('throws a not implemented exception', async () => {
-      await expect(dynamicAuthzService.deleteGroup({ groupId, authenticatedUser: mockUser })).rejects.toThrow(
-        Error
+    let params: DeleteGroupRequest;
+    beforeEach(() => {
+      params = { groupId, authenticatedUser: mockUser };
+      auditAction = 'deleteGroup';
+
+      dynamicAuthzService.getIdentityPermissionsByIdentity = jest.fn().mockResolvedValue({
+        data: {
+          identityPermissions: [{}]
+        }
+      });
+
+      dynamicAuthzService.deleteIdentityPermissions = jest.fn();
+    });
+
+    it('returns the groupID in the data object when the group was successfully deleted', async () => {
+      const mockReturnValue = { data: { groupId } };
+      mockGroupManagementPlugin.deleteGroup = jest.fn().mockResolvedValue(mockReturnValue);
+
+      const response = await dynamicAuthzService.deleteGroup(params);
+
+      expect(response).toMatchObject<DeleteGroupResponse>(mockReturnValue);
+      expect(auditServiceWriteSpy).toHaveBeenCalledWith(
+        {
+          actor: mockUser,
+          source: auditSource,
+          action: auditAction,
+          requestBody: params,
+          statusCode: 200
+        },
+        mockReturnValue
+      );
+    });
+
+    it('throws exception when identity permissions retrieval fails', async () => {
+      const mockReturnValue = new Error();
+      dynamicAuthzService.getIdentityPermissionsByIdentity = jest.fn().mockRejectedValue(mockReturnValue);
+
+      await expect(dynamicAuthzService.deleteGroup(params)).rejects.toThrow(Error);
+
+      expect(auditServiceWriteSpy).toHaveBeenCalledWith(
+        {
+          actor: mockUser,
+          source: auditSource,
+          action: auditAction,
+          requestBody: params,
+          statusCode: 400
+        },
+        mockReturnValue
+      );
+    });
+
+    it('throws exception when group status update fails', async () => {
+      const mockReturnValue = new GroupNotFoundError('Group does not exist.');
+      mockGroupManagementPlugin.setGroupStatus = jest.fn().mockRejectedValue(mockReturnValue);
+
+      await expect(dynamicAuthzService.deleteGroup(params)).rejects.toThrow(GroupNotFoundError);
+
+      expect(auditServiceWriteSpy).toHaveBeenCalledWith(
+        {
+          actor: mockUser,
+          source: auditSource,
+          action: auditAction,
+          requestBody: params,
+          statusCode: 400
+        },
+        mockReturnValue
+      );
+    });
+
+    it('throws RetryError exception when deletion of identity permissions fails', async () => {
```

Comment:
this test is slightly confusing to me. our `mockReturnValue` is a generic `Error` but the test name says it should throw a `RetryError`

Reply:
Per @htztomic any error thrown by `deleteIdentityPermissions` should be wrapped into `RetryError` so the users know that they can retry the operation.